docfx_yaml/extension.py

```diff
@@ -986,6 +986,22 @@ def group_by_package(toc_yaml):
     return new_toc_yaml
 
 
+# Checks names against their uid, if they're different update it.
+def check_name_with_uid(toc_yaml):
+    for entry in toc_yaml:
+        if entry.get("items"):
+            check_name_with_uid(entry['items'])
+
+        # Don't update the name for overview pages or entries without uid
+        if entry['name'] == "Overview" or not entry.get("uid"):
+            continue
+
+        # If there is a discrepancy in the name, use one from uid.
+        name_from_uid = entry['uid'].split(".")[-1]
+        if entry['name'] != name_from_uid:
+            entry['name'] = name_from_uid
```

Comment:
Overall question: why can't we generate `entry` with the right `name` in the first place? I would assume we know the `uid` when we generate the `name`?

Reply:
Ah, I think that was from the side effect of removing `uid` field on the top level entries. This was the 
```
found_node.pop('uid', 'No uid found')
```

line that I removed to ensure `uid` is now always available. Because it wasn't available we had to use this workaround. Now we should be good to use just the `uid` and not have to do all this extra work.

Reply:
This should now be avoided that I've restored `uid` fields on parent entries that's missing `uid` from removing the line below on L1203. I'll update the code accordingly. Thanks for catching this!